docs/upgrade/troubleshooting.md

```diff
@@ -0,0 +1,126 @@
+
+During this stage you can see the upgrade status windows shows:
+
+![](./assets/ts_status_phase2.png){: style="width:60%"}
+
+It will take a while for all nodes to preload images. If the upgrade fails at this phase, the user can check job logs in `cattle-system` namespace:
+
+```
+$ kubectl get jobs -n cattle-system | grep prepare
+apply-hvst-upgrade-9gmg2-prepare-on-node1-with-2bbea1599a-f0e86   0/1           47s        47s
+apply-hvst-upgrade-9gmg2-prepare-on-node4-with-2bbea1599a-041e4   1/1           2m3s       2m50s
+
+$ kubectl logs jobs/apply-hvst-upgrade-9gmg2-prepare-on-node1-with-2bbea1599a-f0e86 -n cattle-system
+...
+```
+
+It's also safe to [start over the upgrade](#start-over-an-upgrade) if an upgrade fails at this phase.
+
+### Phase 3: Upgrade system services
+
+![](./assets/ts_status_phase3.png){: style="width:60%"}
+
+In this phase, Harvester controller upgrades component Helm charts with a job. The user can check the `apply-manifest` job with the following command:
+
+```
+$ kubectl get jobs -n harvester-system -l harvesterhci.io/upgradeComponent=manifest
+NAME                                 COMPLETIONS   DURATION   AGE
+hvst-upgrade-9gmg2-apply-manifests   0/1           46s        46s
+
+$ kubectl logs jobs/hvst-upgrade-9gmg2-apply-manifests -n harvester-system
+...
+```
+
+### Phase 4: Upgrade nodes
+
+![](./assets/ts_status_phase4.png){: style="width:60%"}
+
+The Harvester controller creates jobs on each node (one by one) to upgrade nodes' OSes and RKE2 runtime. For multi-node clusters, there are two kinds of jobs to update a node:
+
+- **pre-drain** job: live-migrate or shutdown VMs on a node. When the job completes, the embedded Rancher service upgrades RKE2 runtime on a node.
+- **post-drain** job: upgrade OS and reboot.
+
+For single-node clusters, there is only one `single-node-upgrade` type job for each node (named with the format `hvst-upgrade-xxx-single-node-upgrade-<hostname>`).
+
+The user can check node jobs by:
+
+```
+$ kubectl get jobs -n harvester-system -l harvesterhci.io/upgradeComponent=node
+NAME                                  COMPLETIONS   DURATION   AGE
+hvst-upgrade-9gmg2-post-drain-node1   1/1           118s       6m34s
+hvst-upgrade-9gmg2-post-drain-node2   0/1           9s         9s
+hvst-upgrade-9gmg2-pre-drain-node1    1/1           3s         8m14s
+hvst-upgrade-9gmg2-pre-drain-node2    1/1           7s         85s
+
+$ kubectl logs -n harvester-system jobs/hvst-upgrade-9gmg2-post-drain-node2
+...
+```
+
+!!! warning
+    Please do not start over an upgrade if the upgrade fails at this phase.
+
+### Phase 5: Clean-up
+
+The Harvester controller deletes the upgrade repository VM and all files that are not needed anymore.
+
+
+## Common operations
+
+### Start over an upgrade
+
+- Log in to a control plane node.
+- List `Upgrade` CRs in the cluster:
```

Comment:
Maybe display as ordered list is better?

Reply:
Fixed